osu.Game.Tournament/Screens/MapPool/MapPoolScreen.cs

```diff
@@ -136,27 +136,40 @@ private void setMode(TeamColour colour, ChoiceType choiceType)
             pickColour = colour;
             pickType = choiceType;
 
-            static Color4 setColour(bool active) => active ? Color4.White : Color4.Gray;
-
             buttonRedBan.Colour = setColour(pickColour == TeamColour.Red && pickType == ChoiceType.Ban);
             buttonBlueBan.Colour = setColour(pickColour == TeamColour.Blue && pickType == ChoiceType.Ban);
             buttonRedPick.Colour = setColour(pickColour == TeamColour.Red && pickType == ChoiceType.Pick);
             buttonBluePick.Colour = setColour(pickColour == TeamColour.Blue && pickType == ChoiceType.Pick);
+
+            static Color4 setColour(bool active) => active ? Color4.White : Color4.Gray;
         }
 
         private void setNextMode()
         {
-            if (CurrentMatch.Value == null)
+            if (CurrentMatch.Value?.Round.Value == null)
                 return;
 
+            int totalBansRequired = CurrentMatch.Value.Round.Value.BanCount.Value * 2;
+
             const TeamColour roll_winner = TeamColour.Red; //todo: draw from match
 
-            var nextColour = (CurrentMatch.Value.PicksBans.LastOrDefault()?.Team ?? roll_winner) == TeamColour.Red ? TeamColour.Blue : TeamColour.Red;
+            var previousBan = CurrentMatch.Value.PicksBans.LastOrDefault()?.Team ?? roll_winner;
+
+            var nextColour = previousBan == TeamColour.Red ? TeamColour.Blue : TeamColour.Red;
+
+            bool hasAllBans = CurrentMatch.Value.PicksBans.Count(p => p.Type == ChoiceType.Ban) >= totalBansRequired;
+
+            if (!hasAllBans)
+                // If it's the third ban or later, we need to check if it's the team's first or second ban in a row
+                nextColour = (CurrentMatch.Value.PicksBans.Count >= 2 ? CurrentMatch.Value.PicksBans[^2]?.Team : previousBan) == TeamColour.Red ? TeamColour.Blue : TeamColour.Red;
+
+            if (hasAllBans && pickType == ChoiceType.Ban)
+            {
+                // When switching from bans to picks, we don't rotate the team colour.
+                nextColour = pickColour;
+            }
 
-            if (pickType == ChoiceType.Ban && CurrentMatch.Value.PicksBans.Count(p => p.Type == ChoiceType.Ban) >= 2)
-                setMode(pickColour, ChoiceType.Pick);
-            else
-                setMode(nextColour, CurrentMatch.Value.PicksBans.Count(p => p.Type == ChoiceType.Ban) >= 2 ? ChoiceType.Pick : ChoiceType.Ban);
+            setMode(nextColour, hasAllBans ? ChoiceType.Pick : ChoiceType.Ban);
```

Comment:
I feel like this is written in a very roundabout and difficult-to-follow manner... I would much prefer something like the following:

```csharp
            var previousColour = CurrentMatch.Value.PicksBans.LastOrDefault()?.Team ?? roll_winner;
            TeamColour nextColour;

            bool hasAllBans = CurrentMatch.Value.PicksBans.Count(p => p.Type == ChoiceType.Ban) >= totalBansRequired;

            if (!hasAllBans)
            {
                // Ban phase.
                // Switch teams every second ban.
                nextColour = CurrentMatch.Value.PicksBans.Count % 2 == 1
                    ? getOppositeTeamColour(previousColour)
                    : previousColour;
            }
            else if (pickType == ChoiceType.Ban)
            {
                // Switching from bans to picks - stay with the last team that was banning.
                nextColour = pickColour;
            }
            else
            {
                // Pick phase.
                // Switch teams every pick.
                nextColour = getOppositeTeamColour(previousColour);
            }

            setMode(nextColour, hasAllBans ? ChoiceType.Pick : ChoiceType.Ban);

            TeamColour getOppositeTeamColour(TeamColour colour) => colour == TeamColour.Red ? TeamColour.Blue : TeamColour.Red;
```

Note that the above is untested. (This PR really should come with tests at this point too.)